From: David Howells <dhowells@redhat.com>
Date: Fri, 20 Sep 2013 14:18:00 +0100
Subject: [PATCH] CacheFiles: Don't try to dump the index key if the cookie has
 been cleared
Git-commit: 509bf24d1810f120cb1e07cb2d30360a79601f71
Patch-mainline: v3.12
References: bnc#880344

Don't try to dump the index key that distinguishes an object if netfs
data in the cookie the object refers to has been cleared (ie.  the
cookie has passed most of the way through
__fscache_relinquish_cookie()).

Since the netfs holds the index key, we can't get at it once the ->def
and ->netfs_data pointers have been cleared - and a NULL pointer
exception will ensue, usually just after a:

	CacheFiles: Error: Unexpected object collision

error is reported.

Signed-off-by: David Howells <dhowells@redhat.com>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Acked-by: NeilBrown <neilb@suse.de>

---
 fs/cachefiles/namei.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.0-SLE11-SP3.orig/fs/cachefiles/namei.c
+++ linux-3.0-SLE11-SP3/fs/cachefiles/namei.c
@@ -57,7 +57,7 @@ void __cachefiles_printk_object(struct c
 		       object->fscache.cookie->parent,
 		       object->fscache.cookie->netfs_data,
 		       object->fscache.cookie->flags);
-		if (keybuf)
+		if (keybuf && cookie->def)
 			keylen = cookie->def->get_key(cookie->netfs_data, keybuf,
 						      CACHEFILES_KEYBUF_SIZE);
 		else
